src/Tribe/Adjacent_Events.php

```diff
@@ -0,0 +1,329 @@
+<?php
+/**
+ * Controls getting a previous or next event from the context of a single event being viewed.
+ */
+
+// Don't load directly
+if ( ! defined( 'ABSPATH' ) ) {
+	die( '-1' );
+}
+
+if ( ! class_exists( 'Tribe__Events__Adjacent_Events' ) ) {
+
+	class Tribe__Events__Adjacent_Events {
+
+		/**
+		 * @var Tribe__Events__Adjacent_Events
+		 */
+		protected static $instance;
+
+		/**
+		 * @var int
+		 */
+		public $current_event_id;
+
+		/**
+		 * @var int
+		 */
+		public $previous_event_id;
+
+		/**
+		 * @var int
+		 */
+		public $next_event_id;
+
+		/**
+		 * @var string
+		 */
+		public $previous_event_link = '';
+
+		/**
+		 * @var string
+		 */
+		public $next_event_link = '';
+
+		/**
+		 * Set the "center" event ID to look on either side of in prev/next methods.
+		 *
+		 * @since TBD
+		 *
+		 * @param int $event_id The event ID to look on either side of in prev/next methods.
+		 */
+		public function __construct( $event_id ) {
+			$this->current_event_id = $event_id;
```

Comment:
Avoid setting it on `__construct`, create a method just for setting the `current_event_id`